### PR TITLE
Hold prototype memory on non-finite observations

### PR DESCRIPTION
**Pull request by Kayvan-Zahiri**
## Summary
- Prototype memory writes an inf observation into an empty slot, then a second inf sample is `inf - inf = NaN` in the EMA residual. NaN distances also win `argmin` as the nearest slot.
- Inf observations versus a finite prototype yield `-inf` logits, so softmax prefers unused classes.
- Fail-closed: skip the memory write when any observation coordinate is non-finite, and return uniform class logits for that query. Later finite samples keep learning from the previous finite prototypes.

## Test plan
- [x] `.venv/bin/python -m pytest tests/test_prototype_memory.py -q`
- [x] `.venv/bin/python -m ruff check alberta_framework/core/prototype_memory.py tests/test_prototype_memory.py`

No Slop measured-run receipt. Made with Cursor.

Made with [Cursor](https://cursor.com)

### CODE REVIEW

**Review by Prabhat1308 — APPROVED** [review]
Independent review of head `b768f6c507c21f7ec7dc4b0b0887751557245a80` against `main` `da8b86c`, under the `review-asi-contributions` skill. Head tree verified as current `main` plus exactly `alberta_framework/core/prototype_memory.py` + `tests/test_prototype_memory.py`; automated adversarial scan clean (no dependency, CI, hook, or lockfile changes; no assertions or raises removed anywhere in the fleet).

**Reproduced:** `tests/test_prototype_memory.py` at head — **9 passed in 1.40s**; repo-wide Ruff: All checks passed!; strict mypy: Success: no issues found in 159 source files; `git diff --check`: clean. Red phase with `alberta_framework/core/prototype_memory.py` reverted to `main`, PR tests kept: **1 failed, 8 passed in 1.16s** (1 new regression test(s) fail), confirming failing-test-first.

**Scope note:** the guard skips updates only on non-finite inputs (or non-finite derived diagnostics); finite paths are numerically unchanged. Later finite samples resume learning from the last finite state.

Process context, stated factually: this PR is part of a 27-PR same-day fleet by one author in the same non-finite-guard class, with honest Cursor authorship disclosure, no measured-run receipt, and no linked pre-registration issue; receipts cannot create score and their absence does not gate correctness — flagged only so the maintainer weighs the fleet, not just each PR.

Recommendation: **accept**.

```slop-review
{"schemaVersion": "1", "projectId": "asi", "artifactUrl": "https://github.com/elizaOS/asi/pull/94", "headSha": "b768f6c507c21f7ec7dc4b0b0887751557245a80", "recommendation": "accept", "reproduced": true, "securityRisk": "none", "duplicateRisk": "none", "usefulArtifacts": ["non-finite guard in alberta_framework/core/prototype_memory.py with failing-test-first regression test(s), reproduced red on main (1 failed, 8 passed in 1.16s) and green at head (9 passed in 1.40s)"], "commands": [".venv/bin/python -m pytest tests/test_prototype_memory.py -q -o addopts=\"\"", ".venv/bin/python -m ruff check .", ".venv/bin/python -m mypy", "git diff --check"], "evidenceUrls": ["https://github.com/elizaOS/asi/pull/94"], "summary": "Head verified as main plus the stated two-file diff; red/green reproduced (1 regression test(s) red on main, 9 passed in 1.40s at head). Part of a 27-PR same-day non-finite-guard fleet, disclosed Cursor-authored, no receipts, no linked issues."}
```
